### PR TITLE
Run recognize_digits and understand_sentiment demo with fault tolerant mode.

### LINKED ISSUE #342: Run recognize_digits and understand_sentiment demo with fault tolerant mode.

### PR DESCRIPTION
**Pull request by wanghaoshuang**
fix #342

### CODE REVIEW

**Comment by wanghaoshuang** [comment]
@Yancey1989 先帮我看下配置是否有问题？先不要approve.

File: demo/recognize_digits/train_ft.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import paddle.v2 as paddle
+import paddle.v2.dataset.common as common
+import os
+import sys
+import glob
+import pickle
+
+
+# NOTE: must change this to your own username on paddlecloud.
+USERNAME = "wanghaoshuang@baidu.com"
```
Comment (typhoonzero) [inline demo/recognize_digits/train_ft.py]: Do not use specific username.

File: demo/recognize_digits/train_ft.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+TRAINER_COUNT = int(os.getenv("PADDLE_INIT_NUM_GRADIENT_SERVERS", "-1"))
+
+def prepare_dataset():
+    # convert will also split the dataset by line-count
+    common.convert(TRAIN_FILES_PATH,
+                paddle.dataset.mnist.train(),
+                8192, "train")
+    common.convert(TEST_FILES_PATH,
+                paddle.dataset.mnist.test(),
+                1, "test")
```
Comment (Yancey1989) [inline demo/recognize_digits/train_ft.py]: `line_count=1` is so small, this will generate too much small files..

**Comment by Yancey1989** [comment]
GPU mode error:
```bash
  File "./train_ft.py", line 169, in <module>
    main()
  File "./train_ft.py", line 117, in main
    predict = convolutional_neural_network(images)
  File "./train_ft.py", line 86, in convolutional_neural_network
    act=paddle.activation.Relu())
  File "/usr/local/lib/python2.7/dist-packages/paddle/trainer_config_helpers/default_decorators.py", line 53, in __wrapper__
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/paddle/trainer_config_helpers/networks.py", line 228, in simple_img_conv_pool
    layer_attr=pool_layer_attr)
  File "/usr/local/lib/python2.7/dist-packages/paddle/trainer_config_helpers/default_decorators.py", line 53, in __wrapper__
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/paddle/trainer_config_helpers/layers.py", line 381, in wrapper
    return method(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/paddle/trainer_config_helpers/layers.py", line 2612, in img_pool_layer
    "only (Cudnn)AvgPooling, (Cudnn)MaxPooling are supported"
AssertionError: only (Cudnn)AvgPooling, (Cudnn)MaxPooling are supported
```

**Pull request merged**